### ISSUE TITLE
Convert initSkipLink.js to TypeScript & add unit tests

### ISSUE THREAD

**Issue by lb-** [wagtail/wagtail#9541]
### Is your proposal related to a problem?

**`initSkipLink` has been in the codebase for some time and it would be good to convert this to TypeScript**

So that we can have a basic confidence that code changes will not lead to widget regressions, and in prep for some widgets migrating to Stimulus it would be good to have a basic test coverage of this component util.

It would be good to also include some basic JSDOC for each function in that file, and conversion to TypeScript gives us more confidence that the code will be used in a reliable way.

### Describe the solution you'd like

The file https://github.com/wagtail/wagtail/blob/main/client/src/includes/initSkipLink.js should be converted to TypeScript (`.ts`) and any required types be added.

Along with this we should add some basic unit tests to cover the init functions' basic behaviour in a new file `client/src/includes/initSkipLink.test.js` (note: .js for tests)

Additionally, there should be a basic JSDOC entry describing the purpose and behaviour of each of the functions in the file - if not already included. For example the 'inspired by' label should be in the JSDOC section.

#### Requirements

* [ ] Convert the file to [TypeScript](https://www.typescriptlang.org/) 
* [ ] Add a [JSDOC](https://jsdoc.app/) description to the function
* [ ] Add a basic set unit test in newly created test file, it should check that the `main` element gains focus in the test and it should check that the tabindex gets correctly added & then removed on blur

### Additional context

* This has been created as part of the Outreachy program to help new contributors that are interested in the Stimulus project.
* Look at client/src/includes/breadcrumbs.test.js for an example of mocking some HTML to then trigger some behaviour on that HTML.
* Look at some future proposed code for an example of unit testing this (note: this is not the same source but should give you a good idea for some starting points) https://github.com/lb-/wagtail/blob/rfcs/stimulus-ui/client/src/controllers/SkipLinkController.test.js

**Comment by Deepanshu039** [comment]
hi, @lb- I'd like to work on this issue.

**Comment by Julietadeboye** [comment]
I am working on this

**Comment by lb-** [comment]
@Julietadeboye as per Slack - yes please can you pick this up. 

Sorry for the confusion @Deepanshu039
